File: Project/PluginDev/lite/base/src/main/java/com/larry/lite/base/LiteConnectionFactory.java
package com.larry.lite.base;

import com.squareup.okhttp.OkHttpClient;
import java.net.HttpURLConnection;
import java.net.MalformedURLException;

public interface LiteConnectionFactory {

    HttpURLConnection openConnection(LitePlugin var1, String var2) throws MalformedURLException;

    OkHttpClient getOkHttpClient();
}

File: Project/PluginDev/lite/base/src/main/java/com/larry/lite/base/LiteConnectionFactory.kt
package com.larry.lite.base

import com.squareup.okhttp.OkHttpClient
import java.net.HttpURLConnection
import java.net.MalformedURLException

interface LiteConnectionFactory {

    @Throws(MalformedURLException::class)
    fun openConnection(litePlugin: LitePlugin, url: String): HttpURLConnection

    val okHttpClient: OkHttpClient
}
